src/test/java/io/nats/client/impl/ConnectionListenerTests.java

```diff
@@ -94,40 +85,32 @@ public void testDisconnectReconnectCount() throws Exception {
             int port;
             try (NatsTestServer ts = new NatsTestServer(false)) {
                 Options options = new Options.Builder().
-                                    server(ts.getURI()).
-                                    reconnectWait(Duration.ofMillis(100)).
-                                    maxReconnects(-1).
-                                    connectionListener(handler).
-                                    build();
+                        server(ts.getURI()).
+                        reconnectWait(Duration.ofMillis(100)).
+                        maxReconnects(-1).
+                        connectionListener(handler).
+                        build();
                 port = ts.getPort();
                 nc = Nats.connect(options);
-                assertTrue(Connection.Status.CONNECTED == nc.getStatus(), "Connected Status");
+                assertSame(Connection.Status.CONNECTED, nc.getStatus(), "Connected Status");
                 assertEquals(ts.getURI(), nc.getConnectedUrl());
                 handler.prepForStatusChange(Events.DISCONNECTED);
             }
 
-            try {
-                nc.flush(Duration.ofMillis(50));
-            } catch (Exception exp) {
-            }
-    
-            handler.waitForStatusChange(400, TimeUnit.MILLISECONDS);
+            try { nc.flush(Duration.ofMillis(250)); } catch (Exception exp) { /* ignored */ }
 
+            handler.waitForStatusChange(1000, TimeUnit.MILLISECONDS);
             assertTrue(handler.getEventCount(Events.DISCONNECTED) >= 1);
             assertNull(nc.getConnectedUrl());
 
-
             try (NatsTestServer ts = new NatsTestServer(port, false)) {
-                try {
-                    Thread.sleep(200);
-                } catch (Exception e) {
-                }
-                assertTrue(Connection.Status.CONNECTED == nc.getStatus(), "Connected Status");
-            
+                try { Thread.sleep(1000); } catch (InterruptedException e) { /* ignored */ }
```

Comment:
give the server enough time to start before testing. Again, the test isn't testing the connection per se (here it's just kind of verifying it's in the right state to continue with the rest of the test)